src/fastify-plugins/maps/static-maps.js

```diff
@@ -0,0 +1,284 @@
+import fp from 'fastify-plugin'
+import { Type as T } from '@sinclair/typebox'
+import asar from '@electron/asar'
+import { Mime } from 'mime/lite'
+import standardTypes from 'mime/types/standard.js'
+
+import { NotFoundError, getFastifyServerAddress } from '../utils.js'
+
+export const PLUGIN_NAME = 'mapeo-static-maps'
+
+export const plugin = fp(staticMapsPlugin, {
+  fastify: '4.x',
+  name: PLUGIN_NAME,
+})
+
+/**
+ * @typedef {object} StaticMapsPluginOpts
+ * @property {string} [prefix]
+ * @property {string} staticRootDir
+ */
+
+/**
+ * @typedef {object} StaticMapsPluginDecorator
+ * @property {(styleId: string, serverAddress: string) => Promise<string>} getResolvedStyleJson
+ * @property {(styleId: string) => Promise<import('node:fs').Stats>} getStyleJsonStats
+ */
+
+/** @type {import('fastify').FastifyPluginAsync<StaticMapsPluginOpts>} */
+async function staticMapsPlugin(fastify, opts) {
+  fastify.decorate('mapeoStaticMaps', {
+    async getResolvedStyleJson(styleId, serverAddress) {
+      const filePath = path.join(opts.staticRootDir, styleId, 'style.json')
+
+      const data = await fs.readFile(filePath, 'utf-8')
+
+      return data.replace(
+        /\{host\}/gm,
+        new URL(`${opts.prefix || ''}/${styleId}`, serverAddress).href
+      )
+    },
+    async getStyleJsonStats(styleId) {
+      const filePath = path.join(opts.staticRootDir, styleId, 'style.json')
+      const stats = await fs.stat(filePath)
+      return stats
+    },
+  })
+
+  fastify.register(routes, {
+    prefix: opts.prefix,
+    staticRootDir: opts.staticRootDir,
+  })
+}
+
+const GetStaticMapTileParamsSchema = T.Object({
+  styleId: T.String(),
+  tileId: T.String(),
+  z: T.Number(),
+  y: T.Number(),
+  x: T.Number(),
+  ext: T.Optional(T.String()),
+})
+
+const ListStaticMapsReplySchema = T.Array(
+  T.Object({
+    id: T.String(),
+    name: T.Union([T.String(), T.Null()]),
+    styleUrl: T.String(),
+  })
+)
+
+const GetStyleJsonParamsSchema = T.Object({
+  styleId: T.String(),
+})
+
+/** @type {import('fastify').FastifyPluginAsync<StaticMapsPluginOpts, import('fastify').RawServerDefault, import('@fastify/type-provider-typebox').TypeBoxTypeProvider>} */
+async function routes(fastify, opts) {
+  const { staticRootDir } = opts
+
+  /**
+   * @param {import('fastify').FastifyRequest<{Params: import('@sinclair/typebox').Static<typeof GetStaticMapTileParamsSchema>}>} req
+   * @param {import('fastify').FastifyReply} rep
+   */
+  async function handleStyleTileGet(req, rep) {
+    const result = getStyleTileInfo(staticRootDir, req.params)
+
+    if (!result) {
+      const { tileId, z, x, y, ext } = req.params
+      throw new NotFoundError(
+        `Tileset id = ${tileId}, ext=${ext}, [${z}, ${x}, ${y}]`
+      )
+    }
+
+    const { data, mimeType, shouldGzip } = result
+
+    if (mimeType) {
+      rep.header('Content-Type', mimeType)
+    }
+
+    if (shouldGzip) {
+      rep.header('Content-Encoding', 'gzip')
```

Comment:
1. Does this actually compress the data?
2. I would think we'd want to do content negotiation (i.e. respect the [Accept-Encoding request header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding)) here.

Reply:
at the moment, no - can probably remove this for now but we may need it at some point (legacy implementation applied gzip for `mvt` and `pbf` sources), but can be addressed later

> I would think we'd want to do content negotiation

Yeah that makes sense - not necessary if we omit this altogether for now though right?

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/435/commits/773a3d65489593bfbd93cb9fe10329747cdbb6d9